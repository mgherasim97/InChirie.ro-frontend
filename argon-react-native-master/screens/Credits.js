import React, { Component, useState } from 'react';
import {
  Modal,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableOpacity,
  View,
  Dimensions,
  ImageBackground,
  Alert
} from 'react-native';
import Carousel from 'react-native-anchor-carousel';
import loadingBlurImage from '../assets/imgs/loading-blur.png';


const { height, width } = Dimensions.get('window');

// const data = [
//   {
//     uri: '../assets/imgs/Luiza_Negru.jpg',
//     title: 'Negru Luiza',
//     content: 'merge cred'
//   },
//   {
//     uri: '../assets/imgs/Daniel_Sava.jfif',
//     title: 'Lorem ipsum ',
//     content: 'Neque porro quisquam est qui dolorem ipsum '
//   },
//   {
//     uri: '../assets/imgs/Mihai_Gherasim.jfif',
//     title: 'Lorem ipsum dolor',
//     content: 'Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...'
//   },
//   {
//     uri: '../assets/imgs/Anca_Stefania_Ilicea.jpg',
//     title: 'Lorem ipsum dolor',
//     content: 'Neque porro quisquam est qui dolorem ipsum quia dolor sit amet'
//   }
// ];

const data = [
  {
    uri: 'https://i.imgur.com/9QlOVC5.jpg',
    title: 'Ilicea Anca Ștefania - "Da-mi bag picioarele!"',
    content: 'Șefa echipei de frontend, nu vrei sa o superi.'
  },
  {
    uri: 'https://i.imgur.com/DN2njLs.jpg',
    title: 'Sava Vasile Daniel - "Noi la MDS când mai lucrăm?"',
    content: 'O minte sclipitoare, coordonator neoficial al proiectului și programator pe backend.'
  },
  {
    uri: 'https://i.imgur.com/yqR5PPp.jpg',
    title: 'Negru Luiza - "Morții lui de laptop!"',
    content: 'Plină de idei, voie bună și dorință de a învăța, programator de backend.'
  },
  {
    uri: 'https://i.imgur.com/VtO3zqE.jpg',
    title: 'Gherasim Mihai - "Acum termin testele... acum..."',
    content: 'Inima echipei, cel mai bun dintre cei mai buni. Sclavul Ancăi pe frontend.'
  }
];

export default class Credits extends Component {
    // constructor(props) {
    //     super(props);
    //     this.modalVisible = false;
    // }
  renderItem = ({ item, index }) => {
    const { uri, title, content } = item;
    
    return (
      <TouchableOpacity
        activeOpacity={1}
        style={styles.item}
        onPress={() => {
          this.numberCarousel.scrollToIndex(index);
        }}
      >
        <ImageBackground
          source={{ uri: uri }}
          style={styles.imageBackground}
          loadingIndicatorSource={loadingBlurImage}
        //   onPress={() => {
        //   this.setState({ modalVisible: !modalVisible })
        // }}
        >
        </ImageBackground>

        <View style={styles.lowerContainer}>
          <Text style={styles.titleText}>{title}</Text>
          <Text style={styles.contentText}>{content}</Text>
        </View>
      </TouchableOpacity>
    );
  };

  render() {
    return (
        <View
        style={styles.carouselContainer}>
            <Carousel
                style={styles.carousel}
                data={data}
                renderItem={this.renderItem}
                itemWidth={0.7 * width}
                inActiveOpacity={0.3}
                containerWidth={width - 10}
                ref={(c) => {
                this.numberCarousel = c;
                }}
            />
        </View>
    );
  }
};

const styles = StyleSheet.create({
  carousel: {
    flex: 1,
    backgroundColor: '#141518'
  },
  carouselContainer: {
    width: width,
    height:height*0.8, 
    marginTop:10
  },
  item: {
    borderWidth: 2,
    backgroundColor: 'white',
    flex: 1,
    borderRadius: 5,
    borderColor: 'white',
    elevation: 3
  },
  imageBackground: {
    flex: 2,
    backgroundColor: '#EBEBEB',
    borderWidth: 1,
    borderColor: 'white'
  }
});