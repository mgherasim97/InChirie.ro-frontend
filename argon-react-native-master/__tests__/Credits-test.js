/**
 * @jest-environment jsdom
 */

import React from 'react';
import Credits from '../screens/Credits';
import { mount, shallow } from 'enzyme';
import {
  StyleSheet,
  Dimensions,
} from 'react-native';

// import jsdom from 'jsdom'
// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
// global.document = doc
// global.window = doc.defaultView

var jsdom = require('jsdom');
const { JSDOM } = jsdom;

const { document } = (new JSDOM('')).window;
global.document = document;

// const React = require('react')
// const Inregistrare = require('../screens/Inregistrare')
// const renderer = require('react-test-renderer')

const { height, width } = Dimensions.get('window');

const styles = StyleSheet.create({
  carousel: {
    flex: 1,
    backgroundColor: '#141518'
  },
  carouselContainer: {
    width: width,
    height:height*0.8, 
    marginTop:10
  },
  item: {
    borderWidth: 2,
    backgroundColor: 'white',
    flex: 1,
    borderRadius: 5,
    borderColor: 'white',
    elevation: 3
  },
  imageBackground: {
    flex: 2,
    backgroundColor: '#EBEBEB',
    borderWidth: 1,
    borderColor: 'white'
  }
});


describe('Image Carousel rendering', () => {

  it('Renders without crashing', () => {

    const component = shallow(<Credits />);
  });

  
  it('Renders correct data', () => {
    const component = shallow(<Credits />);

    expect(component).toMatchSnapshot();
  });


  it('The expected data is shown', () => {
    const component = shallow(<Credits />);

    expect(component.find('Carousel').props().data).toEqual([
        {
            uri: 'https://i.imgur.com/9QlOVC5.jpg',
            title: 'Ilicea Anca Ștefania - "Da-mi bag picioarele!"',
            content: 'Șefa echipei de frontend, nu vrei sa o superi.'
          },
          {
            uri: 'https://i.imgur.com/DN2njLs.jpg',
            title: 'Sava Vasile Daniel - "Noi la MDS când mai lucrăm?"',
            content: 'O minte sclipitoare, coordonator neoficial al proiectului și programator pe backend.'
          },
          {
            uri: 'https://i.imgur.com/yqR5PPp.jpg',
            title: 'Negru Luiza - "Morții lui de laptop!"',
            content: 'Plină de idei, voie bună și dorință de a învăța, programator de backend.'
          },
          {
            uri: 'https://i.imgur.com/VtO3zqE.jpg',
            title: 'Gherasim Mihai - "Acum termin testele... acum..."',
            content: 'Inima echipei, cel mai bun dintre cei mai buni. Sclavul Ancăi pe frontend.'
          }
      ]);

    //component.unmount();
  });


  it('The given style is used', () => {
    const component = shallow(<Credits />);

    expect(component.find('Carousel').props().style).toEqual(styles.carousel);

    // component.unmount();

  });


  it('The correct item width is used', () => {
    const component = shallow(<Credits />);
    
    const { width } = Dimensions.get('window');

    expect(component.find('Carousel').props().itemWidth).toEqual(0.7 * width);

    // component.unmount();

  });


  it('The correct container width is used', () => {
    const component = shallow(<Credits />);
    
    const { width } = Dimensions.get('window');

    expect(component.find('Carousel').props().containerWidth).toEqual(width - 10);

    //component.unmount();
  });
  

  it('The correct container inActiveOpacity is used', () => {
    const component = shallow(<Credits />);

    expect(component.find('Carousel').props().inActiveOpacity).toEqual(0.3);

    //component.unmount();
  });

  it('The correct container style is used', () => {
    const component = shallow(<Credits />);

    expect(component.find('View').props().style).toEqual(styles.carouselContainer);

    component.unmount();
  });
  
});

