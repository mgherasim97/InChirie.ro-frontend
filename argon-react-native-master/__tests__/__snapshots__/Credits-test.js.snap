// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Carousel rendering Renders correct data 1`] = `
<View
  style={
    Object {
      "height": 1067.2,
      "marginTop": 10,
      "width": 750,
    }
  }
>
  <Carousel
    bounces={true}
    containerWidth={740}
    data={
      Array [
        Object {
          "content": "Șefa echipei de frontend, nu vrei sa o superi.",
          "title": "Ilicea Anca Ștefania - \\"Da-mi bag picioarele!\\"",
          "uri": "https://i.imgur.com/9QlOVC5.jpg",
        },
        Object {
          "content": "O minte sclipitoare, coordonator neoficial al proiectului și programator pe backend.",
          "title": "Sava Vasile Daniel - \\"Noi la MDS când mai lucrăm?\\"",
          "uri": "https://i.imgur.com/DN2njLs.jpg",
        },
        Object {
          "content": "Plină de idei, voie bună și dorință de a învăța, programator de backend.",
          "title": "Negru Luiza - \\"Morții lui de laptop!\\"",
          "uri": "https://i.imgur.com/yqR5PPp.jpg",
        },
        Object {
          "content": "Inima echipei, cel mai bun dintre cei mai buni. Sclavul Ancăi pe frontend.",
          "title": "Gherasim Mihai - \\"Acum termin testele... acum...\\"",
          "uri": "https://i.imgur.com/VtO3zqE.jpg",
        },
      ]
    }
    inActiveOpacity={0.3}
    inActiveScale={0.8}
    initialIndex={0}
    itemContainerStyle={Object {}}
    itemWidth={525}
    keyExtractor={[Function]}
    minScrollDistance={25}
    onScrollBeginDrag={[Function]}
    onScrollEnd={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnable={true}
    renderItem={[Function]}
    separatorWidth={0}
    style={
      Object {
        "backgroundColor": "#141518",
        "flex": 1,
      }
    }
  />
</View>
`;
